fewsats/core.py

```diff
@@ -114,6 +121,63 @@ def add_webhook(self:Fewsats,
     return self._request("POST", f"v0/users/webhook/add", json={"webhook_url": webhook_url})
 
 # %% ../nbs/00_core.ipynb 39
+@dataclass
+class FewsatsWebhookEvent:
+    offer_id: str
+    payment_context_token: str
+    amount: int
+    currency: str
+    status: str
+    timestamp: str
+
+@patch(cls_method=True)
+def verify_webhook(cls:Fewsats,
+                   data: bytes,
+                   signature: str,
+                   webhook_secret: str,
+                   ) -> dict:
+    """
+    Verify and parse a webhook that comes from Fewsats
+    Args:
+        data: bytes
+        signature: str
+        webhook_secret: str
+    Returns:
+        FewsatsWebhookEvent
+    """
+    if not isinstance(data, bytes):
+        raise TypeError(f"'data' should be bytes, got {type(data)}")
+
+    timestamp_str, signature_str = signature.split(",")
+    timestamp = timestamp_str.split("=")[1]
+    signature_version, signature = signature_str.split("=")
+
+    payload_str = data.decode("utf-8")
+
+    if signature_version != cls.WEBHOOK_VERSION:
+        raise ValueError("Unsupported signature version")
+
+    if timestamp.isdigit():
+        timestamp = int(timestamp)
+    else:
+        raise ValueError("Invalid timestamp")
+
+    if timestamp - time.time() > 300:
```

Comment:
"is too old" => "Older than 5 minutes ago"  or "is too old: {timestamp}" for having some context